libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m

```diff
@@ -243,6 +245,7 @@ - (void) publishEvents:(NSArray<SFSDKInstrumentationEvent *> *) events {
                 }
                 publishCompleteBlock = nil;
             }
+            [self cleanupBackgroundTask];
```

Comment:
@brandonpage Shouldn't this  clean up occur in the else block above it?

Reply:
Don't we want to end the background task regardless of success/failure? If the operation fails the events aren't deleted and will be sent (with new ones) the next time the app is backgrounded.